### PR TITLE
feat: add stake molecule

### PR DESCRIPTION
**Pull request by wenyuanhust**

### CODE REVIEW

File: contracts/common/types/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub mod generated;
```
Comment (KaoImin) [inline contracts/common/types/src/lib.rs]:
```suggestion
mod generated;

pub use generated::*;
```

File: contracts/common/types/schemas/stake.mol [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import basic;
+
+table StakeInfo {
+	l1_address:       Identity,
+	l2_address:       Identity,
```
Comment (felicityin) [inline contracts/common/types/schemas/stake.mol]: Useless fields.

File: contracts/common/types/schemas/stake.mol [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	amount:           Byte16,
+	inauguration_era: Byte8,
+}
+
+////////////////////
+// stake AT lock
+////////////////////
+table StakeLockArgs {
+	stake_smt_type_id:   Byte32,
+	staker_identity:     Identity,
```
Comment (felicityin) [inline contracts/common/types/schemas/stake.mol]: Too many spaces.

File: contracts/common/types/schemas/stake.mol [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import basic;
+
+table StakeInfo {
+	l1_address:       Identity,
+	l2_address:       Identity,
+	amount:           Byte16,
+	inauguration_era: Byte8,
```
Comment (felicityin) [inline contracts/common/types/schemas/stake.mol]: inauguration_epoch?

**Review by KaoImin — APPROVED** [review]

**Review by felicityin — APPROVED** [review]

**Pull request merged**